verl/utils/checkpoint/megatron_checkpoint_manager.py

```diff
@@ -0,0 +1,269 @@
+        rng_state_path = get_rng_states_checkpoint_path(ckpt_path)
+        print(f"Loading actor rng states from {rng_state_path}")
+        rng_state = torch.load(rng_state_path)
+        # access rng_state for data parallel rank
+        if not use_dist_ckpt:
+            if data_parallel_random_init:
+                rng_state = rng_state[mpu.get_data_parallel_rank()]
+            else:
+                rng_state = rng_state[0]
+        random.setstate(rng_state['random_rng_state'])
+        np.random.set_state(rng_state['np_rng_state'])
+        torch.set_rng_state(rng_state['torch_rng_state'])
+        torch.cuda.set_rng_state(rng_state['cuda_rng_state'])
+        # Check for empty states array
+        if not rng_state['rng_tracker_states']:
+            raise KeyError
+        tensor_parallel.get_cuda_rng_tracker().set_states(rng_state['rng_tracker_states'])
+
+    def load_checkpoint(self, local_path: str, hdfs_path: str = None, del_local_after_load=False):
+        local, ckpt_path = self.checkpath(local_path, hdfs_path)
+
+        if ckpt_path is None:
+            return
+
+        if 'model' in self.checkpoint_contents:
+            model_path = get_model_checkpoint_path(ckpt_path)
+            ckpt_name = self.get_checkpoint_name(model_path, return_base_dir=False)
+            state_dict = torch.load(os.path.join(ckpt_name))
+            self.weight_loader(state_dict,
+                               self.model,
+                               self.hf_config,
+                               self.param_dtype,
+                               is_value_model=self.is_value_model,
+                               tie_word_embeddings=self.share_embeddings_and_output_weights)
+
+        if 'optimizer' in self.checkpoint_contents:
+            self.load_optimizer(ckpt_path)
+
+        if 'extra' in self.checkpoint_contents:
+            self.load_rng_states(ckpt_path)
+
+        if del_local_after_load:
+            try:
+                os.remove(local_path) if is_non_local(local_path) else None
+            except Exception as e:
+                print(
+                    f'[rank-{self.rank}]: remove local resume ckpt file after loading failed, exception {e} will be ignored'
+                )
+
+    def save_checkpoint(self, local_path: str, hdfs_path: str = None, global_step: int = 0, remove_previous_ckpt=False):
+        local, ckpt_path = self.checkpath(local_path, hdfs_path)
+
+        # record the previous global step
+        self.previous_global_step = global_step
+
+        # remove previous local_path
+        # TODO: shall we remove previous ckpt every save?
+        if remove_previous_ckpt:
+            self.remove_previous_save_local_path()
+        if local:
+            local_path = self.local_mkdir(local_path)
+        torch.distributed.barrier()
+
+        # Save Model
+        if 'model' in self.checkpoint_contents:
+            state_dict = self.weight_saver(self.model,
```

Comment:
It seems that we store the whole weight. We should save the sharded weight here for better resume